HeadersImports.md

```diff
@@ -0,0 +1,68 @@
+# Headers and Imports
+
+## Introduction
+
+Follow this set of guidelines when creating header files and importing them. The
+guidelines are designed to support a wide range of build systems and usage scenarios.
+
+In this document, the term `library` refers to a buildable package. In CocoaPods, it's a CocoaPod.
+In Swift Package Manager, it's a library target.
+
+## Header File Types and Locations
+
+* *Public Headers* - Headers that define the library's API. They should be located in
+  `FirebaseFoo/Sources/Public`. Any additions require a minor version update. Any changes or
+  deletions require a major version update.
+
+* *Public Umbrella Header* - A single header that includes the full library's public API located at
+  `FirebaseFoo/Sources/Public/FirebaseFoo.h`. This header should be included in
```

Comment:
Minor nit: even though we're talking about headers and they are included by the compiler, "included in" still reads as if you're talking about copying it into `Firebase.h`, when you really mean a new reference should be added.

Consider "included from" or rephrasing to more directly/concretely state what you want people to do, e.g.

> Add a new `__has_include` stanza for your umbrella header to Firebase.h.

Also, this is written as if this is unconditional, but really this should only apply to new products, right? For example, when adding GoogleDataTransport, we shouldn't add it to `Firebase.h`, right?

Reply:
I clarified and moved the `Firebase.h` info to `AddNewPod.md`.